05_timers/kernel_space/kernel_time.c

```diff
@@ -0,0 +1,91 @@
+// SPDX-License-Identifier: GPL
+/*
+ * Copyright (c) 2021 Vasyl Yosypenko
+ */
+
+#include <linux/init.h>
+#include <linux/kernel.h>
+#include <linux/kobject.h>
+#include <linux/module.h>
+
+#define GET_ENTIRE_SEC(TIME_NS) (TIME_NS / (1000000000UL))
+#define GET_REMAIN_NS(TIME_NS) (TIME_NS % (1000000000UL))
+
+static struct kobject *time_kobj;
+
+static ssize_t show_abs_time(struct kobject *kobj, struct kobj_attribute *attr, char *buf);
+static ssize_t show_rel_time(struct kobject *kobj, struct kobj_attribute *attr, char *buf);
+
+
+static struct kobj_attribute abs_time_attribute =
+    __ATTR(abs_time, 0664, show_abs_time, NULL);
+
+static struct kobj_attribute rel_time_attribute =
+    __ATTR(rel_time, 0664, show_rel_time, NULL);
+
+static ssize_t show_abs_time(struct kobject *kobj, struct kobj_attribute *attr, char *buf)
+{
+    static u64 last_abs_time;
+
+    sprintf(buf, "%llu.%llu sec\n", GET_ENTIRE_SEC(last_abs_time), GET_REMAIN_NS(last_abs_time));
```

Comment:
Mistake in format string for nanoseconds.

Reply:
Fixed.